modules/internal/DefaultRectangular.chpl

```diff
@@ -415,17 +422,21 @@ module DefaultRectangular {
       if stridable {
         type strType = chpl__signedType(idxType);
         for param i in 1..rank {
-          const rStride = ranges(i).stride:strType,
-                fStride = followThis(i).stride:strType;
-          if ranges(i).stride > 0 {
-            const low = ranges(i).alignedLow + followThis(i).low*rStride,
-                  high = ranges(i).alignedLow + followThis(i).high*rStride,
-                  stride = (rStride * fStride):idxType;
+          // See domain follower for comments about this
+          const rStride = ranges(i).stride;
+          const rSignedStride = rStride:strType,
+                fSignedStride = followThis(i).stride:strType;
+          if rStride > 0 {
+            const riStride = rStride:idxType;
+            const low = ranges(i).alignedLow + followThis(i).low*riStride,
+                  high = ranges(i).alignedLow + followThis(i).high*riStride,
+                  stride = (rSignedStride * fSignedStride):strType;
             block(i) = low..high by stride;
           } else {
-            const low = ranges(i).alignedHigh + followThis(i).high*rStride,
-                  high = ranges(i).alignedHigh + followThis(i).low*rStride,
-                  stride = (rStride * fStride):idxType;
+            const irStride = (-ranges(i).stride):idxType;
```

Comment:
oh yeah - that was an oversight. I meant to, so I will fix. Thanks for noticing!